### ISSUE TITLE
Package request: Wolfram

### ISSUE THREAD

**Issue by arnoudbuzing** [NixOS/nixpkgs#347090]
**Project description**

There is a package for "Mathematica", but as of version 14.1 should be packaged on NixOS as "Wolfram" (a unified app)

The motivation for this can be found here:

https://writings.stephenwolfram.com/2024/07/yet-more-new-ideas-and-new-functions-launching-version-14-1-of-wolfram-language-mathematica/#theres-now-a-unified-wolfram-app

**Metadata**

* homepage URL: https://www.wolfram.com
* source URL: https://user.wolfram.com/portal/myProducts.html
* license: non-free
* platforms: mac/windows/linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by doronbehar** [comment]
Hmm I think the correct thing to do is to simply use an alias. We do have support for this `Wolfram_14.1.0_LINUX.sh`:

https://github.com/NixOS/nixpkgs/blob/5509aba82576ed57b1d2f7dca30e9e02637d329f/pkgs/applications/science/math/mathematica/versions.nix#L10-L16

cc @herberteuler @rafaelrc7 @chewblacka

I personally think that this new name isn't yet so widely used that it justifies even an alias. Even in the website it is still named in some places as "Mathematica". Perhaps in a year or so from now?

**Comment by herberteuler** [comment]
Implementation-wise, the current installer will be reused until `Wolfram_*.sh` becomes incompatible with `Mathematica_*.sh`. This means `generic.nix` can be left untouched.

To maintain conceptual consistency in the naming, the package definition (`default.nix`) can be modified to generate names depending on the version: `mathematica` for version `<= 14.0` and `wolfram` for version `>= 14.1`. However, `mathematica` for version `>= 14.1` probably should be kept for some time due to the great influence of the existing name.

What do you think about this idea?

**Comment by arnoudbuzing** [comment]
I think you should prepare to transition to using 'wolfram' instead of 'mathematica'. At the level of package managers, using 'wolfram' is more correct now, and 'mathematica' is a specific product/licensing configuration inside 'wolfram'.

**Comment by herberteuler** [comment]
I just created a draft. What do you think about it?

**Comment by StrangeGirlMurph** [comment]
I am not sure I understand. How does one even install 14.1.0 now? My uni only provides a `Wolfram_14.1.0_LIN.iso` and not a `Mathematica_13.0.1_BNDL_LINUX.sh` anymore since 13.0.1 which contains: 
```
~/Downloads/Wolfram_14.1.0_LIN
» tre
.
└── Unix
    ├── Installer
    │   ├── WolframInstaller
    │   └── TextResources
    │       ├── English
    │       │   ├── Prompt
    │       │   ├── Text
    │       │   ├── Error
    │       │   └── Verbose
    │       └── Greeting.English
    └── Files
        ├── Core.W-LINUX-Standard
        │   ├── contents.tar.xz
        │   └── info
        ├── FunctionInformation.W-LINUX-Standard
        │   ├── contents.tar.xz
        │   └── info
        ├── Paclets.W-LINUX-Standard
        │   ├── contents.tar.xz
        │   └── info
        └── Documentation.W-LINUX-Standard
            ├── contents.tar.xz
            └── info
```

If I just try to use the `WolframInstaller` with the current setup:
`nix-store --query --hash $(nix store add-path ./Unix/Installer/WolframInstaller --name 'Mathematica_14.1.0_BNDL_LINUX.sh')`
```nix
my_mathematica = pkgs.mathematica.override {
    source = pkgs.requireFile {
      name = "Mathematica_14.1.0_BNDL_LINUX.sh";
      sha256 = "1rhar418k9lv9r5yzjprr6pybmnhhpirbl1ryn5rl8jziyrk93s5";
      message = ''
        Your override for Mathematica includes a different src for the installer, and it is missing.
      '';
      hashMode = "recursive";
    };
  };
```

It just doesn't work :( Help!

**Comment by doronbehar** [comment]
@StrangeGirlMurph Are you saying that your university is only providing an ISO and not a `.sh` file? That's very peculiar and indeed we don't have support for this, (but it'd be nice to support it). I'd try to play with that ISO on a different Linux distribution (perhaps use distrobox) and see how is this file supposed to be used.

Once you are capable of installing ~~Mathematica~~ Wolfram from that ISO without manual interruption, this achievement could be contributed and shared in Nixpkgs.

In any case, no matter what is the Name of the package and/or attribute - Mathematica or Wolfram, your challenge is out of scope for this issue.

**Comment by StrangeGirlMurph** [comment]
Yes we only get the .iso and it only contains the WolframInstaller script.

**Comment by arnoudbuzing** [comment]
Hi, I work for Wolfram and may be able to help ... Please connect with me on the Wolfram discord server: https://discord.gg/Cq5CybpT

**Comment by SandaruKasa** [comment]
This is not a package request per se, it's more of a renaming request.

Maintainer ping @rafaelrc7
